exercise-analyst-server/API/Services/MeasurementsDev/Dtos/Responses/GetMeasurementsResponse.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System.Collections.Generic;
+
+namespace API.Services.MeasurementsDev.Dtos.Responses
+{
+    public class GetMeasurementsResponse
+    {
+        public List<MeasurementForGetMeasurementsCsvResponse> Measurements { get; set; }
+    }
+
+    public class MeasurementForGetMeasurementsCsvResponse
+    {
+        public int Id { get; set; }
+        public string Activity { get; set; }
+        public int IdFromMobile { get; set; }
+        public int Repetitions { get; set; }
+        public List<AccelerometerMeasurementForGetMeasurementsCsvResponse> AccelerometerMeasurements { get; set; }
+        public List<GyroscopeMeasurementForGetMeasurementsCsvResponse> GyroscopeMeasurements { get; set; }
+    }
+
+    public class GyroscopeMeasurementForGetMeasurementsCsvResponse
+    {
+        public long Id { get; set; }
+        public long TimestampUtc { get; set; }
+        public double X { get; set; }
```

Comment:
To jest chyba dobry moment na poruszenie tego tematu, dostajemy dane w jednym formacie (z użyciem `Vector`) przechowujemy je w innym i wysyłamy je w zasadzie w takim jak przechowujemy. Czy powinniśmy się starać żeby to było bardziej ujednolicone? Ważne imho bo wczoraj się z godzinę zastanawiałem co z tym zrobić i czuje, że zaraz jedna strona będzie chciała tak a inna inaczej a my utkniemy z jednym kontrolerem od wszystkiego.

Reply:
Rozmawiałem z MLem i sami do końca nie wiedzą czego chcą. Chcieli jakiegoś CSVka, którego jednak nie sposób w sensowny sposób zrobić.

Chcieli te info co mamy, czyli nazwe aktywności fizycznej, id pomiaru, timestampy i X, Y i Z od żyroskopu i akcelerometru, czyli właściwie to co mamy.
Tylko do CSV nie bardzo jest to jak zapisać. Jeśli chcieliby tylko żyroskop albo tylko akcelerometr + ogólne info o aktywności to nie ma problemu, ale problem pojawia sięw momencie gdy chcemy dać zarówno akcelerometr i żyroskop jednocześnie. 

Próbowaliśmy kombinować z timestampami, ale te z żyroskopu nie pokrywają się z akcelerometru.
Próbowaliśmy więc zaokrąglać timestampy tak, żeby mieć dopasowanie 1:1 do pomiaru z żyroskopu i akcelerometru, ale niestety to też jest kiepskie rozwiązanie, bo zaokrąglajac do nawet najmniejszych części dostajemy 3-5 pomiary, więc żeby je zgrupować po timestampach musielibyśmy albo uśredniać pomiary np. z żyroskopu (których nota bene jest więcej, niż z akcelerometru, tak widocznie działa ten czujnik), żeby zredukować proporcję 1:1, ale to jest takie już gmeranie w danych i tracenie precyzji, co jeśli zrobimy na serwerze to już tego nie odkręcą.

Wobec czego zdecydowaliśmy z MLem, że dostaną po prostu ustrukturyzowanego JSONa i zobaczą co z tym da się zrobić. Jest tam trochę osób (chyba 8), może ktoś coś ogarnie, bo do końca też nie wiedzą jak miałby wyglądać ten CSV, który początkowo chcieli.

No, a jak tutaj im utnę precyzję timestampów lub nawale redundantnych wpisów, to potem albo tego nie odfiltrują ze spamu, albo nie utracą precyzję, która być może jest istotna - tu już sami będą lepiej wiedzieć czego naprawdę chcą, z poziomu pythona można jak najbardziej obrobić jsona.

Też się nad tym głowiłem i robiłem rózne POC ale do niczego lepszego nie doszedłem.

Reply:
Koncept wysyłania csv bym olał, jestesmy serwerem http wysyłamy json/xml a cache imho to mogą zrobić nawet w protobufie byle tylko po swojej stronie.
Tbh po przemyśleniu tej spójności requestów i responsów to proponuje przerobić ten create co ETL dostał na taki format jak masz tutaj (w najgorszym wypadku wrzuci się do `api/v2`). Mniej roboty z mapami i mamy elegancki dostęp do modelu

Reply:
Jak dla mnie to może zostać jak teraz jest i tyle, myślę, że nie ma co overthinkować co tutaj się za dzieje 😃